src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -838,6 +838,8 @@ public synchronized int loadNewSSTablesWithCount(boolean assumeCfIsEmpty)
                 continue; // old (initialized) SSTable found, skipping
             if (descriptor.type.isTemporary) // in the process of being written
                 continue;
+            if (entry.getValue().contains(Component.OBSOLETE))
+                continue;
```

Comment:
is the implication here that by continuing, we're not initializing the sstable and we *will* delete it later? or do we need to update something elsewhere? what ensure "tmp" sstables get deleted?

Reply:
also thoughts on if it's worth us logging (either by service log or metrics) some info about the fact that we've encountered an "obsolete" sstable (either here or later when we would delete it...) for transparenty/debuggability in the future

Reply:
this section runs in parallel with the scrubDataDirectories code that does the actual deletion unfortunately https://github.com/palantir/cassandra/pull/521/files#diff-9db147d2e310e83ba17e4b4c756ce0aa90ce1742006fa3bc2418aaaf1fe4b7c9L573-L611 which is why i assume descriptor.type.isTemporary tables are also skipped here (they're also scrubbed in scrubDataDirectories). without this part, you get concurrent deletion issues like in these test failures: https://app.circleci.com/pipelines/github/palantir/cassandra/3931/workflows/00830712-dd20-49e5-a02f-0b2f70fb3429/jobs/11440/tests

Reply:
added warn log when the scrubber encounters an obsolete sstable

Reply:
👍🏻